core/main_test.ts

```diff
@@ -1260,6 +1260,453 @@ actions:
+        }
+        select * from \${ref({name: "A", includeDependentAssertions: true})}
+        select * from \${ref({name: "C", includeDependentAssertions: false})}
+        select 1 as btest
+        `);
+        fs.writeFileSync(path.join(projectDir, "definitions/C.sqlx"),
+          `config {
+            type: "table",
+             assertions: {
+               rowConditions: ["test > 1"]
+            }
+          }
+          SELECT 1 as test
+        }
+        `);
+
+        const result = runMainInVm(coreExecutionRequestFromPath(projectDir));
+        expect(result.compile.compiledGraph.graphErrors.compilationErrors.length).deep.equals(2);
+        expect(result.compile.compiledGraph.graphErrors.compilationErrors[0].message).deep.equals(`Conflicting "includeDependentAssertions" properties are not allowed. Dependency A has different values set for this property.`);
+      })
+
+      suite("Action configs", ({ beforeEach }) => {
+        let projectDir: any;
+
+        beforeEach("Create temporary dir and files", () => {
+          projectDir = tmpDirFixture.createNewTmpDir();
+          fs.writeFileSync(
+            path.join(projectDir, "workflow_settings.yaml"),
+            dumpYaml(dataform.WorkflowSettings.create(testConfig))
+          );
+          fs.mkdirSync(path.join(projectDir, "definitions"))
+          fs.writeFileSync(path.join(projectDir, "definitions/A.sql"), "SELECT 1 as test");
+          fs.writeFileSync(path.join(projectDir, "definitions/A_assert.sql"), "SELECT test from A");
+          fs.writeFileSync(path.join(projectDir, "definitions/B.sql"), "SELECT 1");
+          fs.writeFileSync(path.join(projectDir, "definitions/C.sql"), "SELECT 1");
+          fs.writeFileSync(
+            path.join(projectDir, `definitions/notebook.ipynb`),
+            EMPTY_NOTEBOOK_CONTENTS
+          );
+        });
+
+        test(`When dependOnDependencyAssertions property is set to true, assertions from A are added as dependencies`, () => {
+          fs.writeFileSync(
+            path.join(projectDir, "definitions/actions.yaml"),
+            `
+            actions:
+              - table:
+                  filename: A.sql
+              - assertion:
+                  filename: A_assert.sql
+                  dependencyTargets:
+                    - name: A
+              - view:
+                  filename: B.sql
+                  dependOnDependencyAssertions: true
+                  dependencyTargets:
+                    - name: A
+              - operation:
+                  filename: C.sql
+                  dependOnDependencyAssertions: true
+                  dependencyTargets:
+                    - name: A
+              - notebook:
+                  filename: notebook.ipynb
+                  dependOnDependencyAssertions: true
+                  dependencyTargets:
+                    - name: A
```

Comment:
I much prefer having file contents unindented, as the current tests in the file are - it prevents confusing diff errors when calling `expect`.

In any case, we should be consistent.